Add default values to external APIs

ehashman opened the pull request:
```release-note
Set some default values (e.g. RouterProfile) when not specified upon creating a new cluster config
```
Addresses #336.

- [x] Default RouterProfiles.
- [ ] Default masterPoolProfile/count=3
- [ ] Default agentPoolProfiles[name=infra]/count=3
- [ ] Default agentPoolProfiles[*]/osType=Linux
- [x] Unit test defaults
- [ ] Simplify test/manifests/fakerp/create.yaml

ehashman commented:
/retest

flake #1549

ehashman commented:
/retest

kwoodson commented on pkg/api/2018-09-30-preview/default.go:
```diff
@@ -0,0 +1,13 @@
+package v20180930preview
+
+func setDefaults(oc *OpenShiftManagedCluster) {
+	defaultRouterName := "default"
```
default router name is `router`

ehashman replied:
Hm, are you sure? I got this from https://github.com/openshift/openshift-azure/blob/a4338bc5aeaff48ba2a37bec28ae249231f47dd8/pkg/fakerp/fakerp.go#L275-L281

I'd be concerned if I'm setting the wrong name here while all the tests are passing...

jim-minter replied:
@ehashman you're right

kwoodson replied:
I thought this was referring to the actual router name instead of the routerprofile name.

kwoodson commented on pkg/api/2019-04-30/default.go:
```diff
@@ -0,0 +1,13 @@
+package v20190430
+
+func setDefaults(oc *OpenShiftManagedCluster) {
+	defaultRouterName := "default"
```
same as above

mjudeikis commented:
/retest

jim-minter commented on pkg/api/2018-09-30-preview/default.go:
```diff
@@ -0,0 +1,13 @@
+	if len(oc.Properties.RouterProfiles) == 0 {
+		oc.Properties.RouterProfiles = []RouterProfile{
+			{
+				Name: &defaultRouterName,
```
in our codebase prefer using to.StringPtr() to skip the extra variable definition

jim-minter commented:
good start, cross-check against https://github.com/openshift/openshift-azure/issues/336 and https://github.com/openshift/openshift-azure/blob/master/test/manifests/fakerp/create.yaml for other obvious defaultables.  I think the following are certainly plausible:

masterPoolProfile/count=3
agentPoolProfiles[name=infra]/count=3
agentPoolProfiles[*]/osType=Linux

+ unit test
+ simplify /test/manifests/fakerp/create.yaml

ehashman commented:
@jim-minter: added your suggestions to a TODO list in the first comment.

In 5908786f2a20cf3195035efed508e22210c0601d I added unit tests and that's taken the PR size from medium to large; perhaps we should split the remaining items into follow-up PRs? Namely, one for more defaults for the master/agentpool profiles, and another for simplifying the test manifest. (If you agree, feel free to remove the WIP label.)

ehashman commented:
Will split the other TODOs into two follow-up PRs.

/retest
